common/src/verify.rs

```diff
@@ -372,63 +372,46 @@ impl CommitSequenceVerifier {
-                    last_extra_agenda_timestamp,
-                },
-            ) => {
-                match tx {
-                    ExtraAgendaTransaction::Delegate(tx) => {
-                        // Update reserved reserved_state by applying delegation
-                        self.reserved_state.apply_delegate(tx).map_err(|e| {
-                            Error::InvalidArgument(format!("invalid delegation: {e}"))
-                        })?;
-                        // Check if extra-agenda transactions are in chronological order
-                        if tx.data.timestamp < *last_extra_agenda_timestamp {
-                            return Err(Error::InvalidArgument(
-                                format!("invalid extra-agenda transaction timestamp: expected larger than or equal to the last transaction timestamp {}, got {}", last_extra_agenda_timestamp, tx.data.timestamp)
-                            ));
+                    Report(_tx) => unimplemented!(),
+                };
+                // Phases
+                use Phase::*;
+                match phase {
+                    AgendaProof { .. } => {
+                        // Continue the phase
+                        self.phase = ExtraAgendaTransaction {
+                            last_extra_agenda_timestamp: timestamp,
                         }
-                        *last_extra_agenda_timestamp = tx.data.timestamp;
                     }
-                    ExtraAgendaTransaction::Undelegate(tx) => {
-                        // Update reserved reserved_state by applying undelegation
-                        self.reserved_state.apply_undelegate(tx).map_err(|e| {
-                            Error::InvalidArgument(format!("invalid undelegation: {e}"))
-                        })?;
+                    ExtraAgendaTransaction {
+                        last_extra_agenda_timestamp,
+                    } => {
                         // Check if extra-agenda transactions are in chronological order
-                        if tx.data.timestamp < *last_extra_agenda_timestamp {
+                        if timestamp < *last_extra_agenda_timestamp {
                             return Err(Error::InvalidArgument(
-                                format!("invalid extra-agenda transaction timestamp: expected larger than or equal to the last transaction timestamp {}, got {}", last_extra_agenda_timestamp, tx.data.timestamp)
+                                format!("invalid extra-agenda transaction timestamp: expected larger than or equal to the last transaction timestamp {last_extra_agenda_timestamp}, got {timestamp}")
                             ));
                         }
-                        *last_extra_agenda_timestamp = tx.data.timestamp;
+                        *last_extra_agenda_timestamp = timestamp;
                     }
-                    ExtraAgendaTransaction::Report(_tx) => unimplemented!(),
+                    _ => unreachable!(),
```

Comment:
It is reachable. Note that you are missing `PhaseMismatch`.
I don't get the point of your refactoring.